posh-git.psm1

```diff
@@ -33,83 +33,71 @@ else {
-        if ($curPath -and $curPath.ToLower().StartsWith($Home.ToLower()))
+        $currentPath = if ($pathInfo.Drive) { $pathInfo.Path } else { $pathInfo.ProviderPath }
+
+        # File system paths are case-sensitive on Linux and case-insensitive on Windows and macOS
+        if (($PSVersionTable.PSVersion.Major -ge 6) -and $IsLinux) {
+            $stringComparison = [System.StringComparison]::Ordinal
+        }
+        else {
+            $stringComparison = [System.StringComparison]::OrdinalIgnoreCase
+        }
+
+        # Abbreviate path by replacing beginning of path with ~ *iff* the path is in the user's home dir
+        if ($currentPath -and $currentPath.StartsWith($Home, $stringComparison))
         {
-            $curPath = "~" + $curPath.SubString($Home.Length)
+            $currentPath = "~" + $currentPath.SubString($Home.Length)
         }
 
-        # Write the current path.
-        Write-Host $curPath -NoNewline
+        # Write the abbreviated current path
+        Write-Host $currentPath -NoNewline
 
-        # Write the Git status summary information.
+        # Write the Git status summary information
         Write-VcsStatus
 
-        # When in debug mode, let user know
-        if ((Test-Path Variable:\PSDebugContext) -or [runspace]::DefaultRunspace.Debugger.InBreakpoint) {
-            $promptSuffix = "`n[DBG]: PS>> "
-        }
-        else {
-            $promptSuffix = "`nPS> "
-        }
+        # If stopped in the debugger, the prompt needs to indicate that in some fashion
+        $debugMode = (Test-Path Variable:/PSDebugContext) -or [runspace]::DefaultRunspace.Debugger.InBreakpoint
+        $promptSuffix = if ($debugMode) { $GitPromptSettings.PromptDebugSuffix } else { $GitPromptSettings.PromptSuffix }
 
         $global:LASTEXITCODE = $origLastExitCode
         $promptSuffix
-    }
+'@)
 
-    # Install the posh-git prompt as the default prompt
+    # Set the posh-git prompt as the default prompt
     Set-Item Function:\prompt -Value $poshGitPromptScriptBlock
 
-    # If default posh-git prompt is two lines then we should ensure PSReadline is set to handle that
-    if (Get-Command Set-PSReadlineOption -ErrorAction SilentlyContinue) {
-        try {
-            $origPsrlExtraPromptLineCount = (Get-PSReadlineOption).ExtraPromptLineCount
-            Set-PSReadlineOption -ExtraPromptLineCount 1 -ErrorAction SilentlyContinue
-        }
-        catch {
-            Write-Debug "Failed to get or set PSReadline ExtraPromptLineCount. Error: $_"
-        }
-    }
-
     $promptReplaced = $true
 }
 
 # Install handler for removal/unload of the module
 $ExecutionContext.SessionState.Module.OnRemove = {
-    $Global:VcsPromptStatuses = $Global:VcsPromptStatuses | Where-Object { $_ -ne $PoshGitVcsPrompt }
+    $global:VcsPromptStatuses = $global:VcsPromptStatuses | Where-Object { $_ -ne $PoshGitVcsPrompt }
 
     if ($promptReplaced) {

```

Comment:
Wondering if this flag has been made obsolete by the `$promptDef -eq $poshGitPromptScriptBlock` check? I suppose it's an optimization, but I wouldn't be opposed to reducing complexity at the expense of a marginally slower uninstall.